appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/ExecutorServicesImpl.java

```diff
@@ -65,12 +70,16 @@
 
     private final ExecutorService taskExecutor;
     private final ContextualTimerExecutor timerExecutor;
-    private PayaraExecutorService executor;
+    private final ManagedExecutorService executor;
+    private final org.glassfish.concurrent.config.ManagedExecutorService config;
     
-    public ExecutorServicesImpl(PayaraExecutorService service) {
-        executor = service;
+    public ExecutorServicesImpl() throws NamingException {
+        InitialContext ctx = new InitialContext();
+        executor = (ManagedExecutorService) ctx.lookup("java:comp/DefaultManagedExecutorService");
         taskExecutor = new ContextualTaskExecutor();
         timerExecutor = new ContextualTimerExecutor();
+        this.config = Globals.getDefaultHabitat()
+                .getService(org.glassfish.concurrent.config.ManagedExecutorService.class);
```

Comment:
You may have multiple services created explicitly, so this might have different behaviour when domain has some executor services created under explicit names.